pandoc_mermaid_filter.py

```diff
@@ -38,13 +40,15 @@ def mermaid(key, value, format_, _):
                 if os.path.isfile('.puppeteer.json'):
                     cmd.extend(["-p", ".puppeteer.json"])
 
-                subprocess.check_call(cmd)
+                subprocess.check_call(cmd, timeout=120)
                 sys.stderr.write('Created image ' + dest + '\n')
 
             return Para([Image([ident, [], keyvals], caption, [dest, typef])])
 
 
 def main():
+    flags = fcntl.fcntl(sys.stdout, fcntl.F_GETFL)
+    fcntl.fcntl(sys.stdout, fcntl.F_SETFL, flags&~os.O_NONBLOCK)
```

Comment:
Can you elaborate on this?

Reply:
when encountering a 
```
BlockingIOError: [Errno 11] write could not complete without blocking
```
this seemed to be the appropriate answer. 

The first part [builds up the file status flags](https://www.gnu.org/software/libc/manual/html_node/Getting-File-Status-Flags.html) for sys.stdout. 
The second part sets the status flags [adding os.O_NONBLOCK](https://docs.python.org/3/library/os.html#os.O_NONBLOCK) to prevent a long-ish mermaid task from halting everything.

Given that this is a Unix specific flag, I guess one could consider the case that this might be running on MS Windows and add appropriate support?